Library/Homebrew/utils/curl.rb

```diff
@@ -303,6 +308,85 @@ def curl_http_content_headers_and_checksum(url, specs: {}, hash_needed: false, u
     def http_status_ok?(status)
       (100..299).cover?(status.to_i)
     end
+
+    # Separates the output text from `curl` into an array of response heads and
+    # the final response body.
+    # @param output [String] The output text from `curl` containing
+    #   response head(s), body, or both.
+    # @return [Hash] A hash containing an array of the response heads and the
+    #   body output, if found.
+    def parse_curl_output(output)
+      heads = []
+      return { heads: heads, body: "" } unless output.is_a?(String)
+
+      output = output.lstrip
+      while output.match?(%r{\AHTTP/[\d.]+ \d+})
+        head_text, _, output = output.partition("\r\n\r\n")
+        output = output.lstrip
+        next if head_text.blank?
+
+        head_text.chomp!
+        head = parse_curl_head(head_text)
+        heads << head if head.present?
+      end
+
+      { heads: heads, body: output }
+    end
+
+    # Parses a `curl` response head into a hash containing the status
+    # information and headers.
+    # @param head_text [String] The head text of a `curl` response.
+    # @return [Hash] A hash containing the status information and headers
+    #   (as a hash with header names as keys).
+    def parse_curl_head(head_text)
```

Comment:
This seems to only be used once so I'd suggest it may be better moved from `utils`.